src/overlays/actors/ovl_Obj_Taru/z_obj_taru.c

```diff
@@ -28,55 +36,306 @@ const ActorInit Obj_Taru_InitVars = {
+        rotY = (Rand_Next() >> 0x11) + this->dyna.actor.yawTowardsPlayer + 0xC000;
+        spawnedActor = Actor_Spawn(&play->actorCtx, play, ACTOR_EN_SW, this->dyna.actor.world.pos.x,
+                                   this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, 0, rotY, 0, params1F);
+        if (spawnedActor != NULL) {
+            spawnedActor->parent = &this->dyna.actor;
+            spawnedActor->velocity.y = 12.0f;
+            spawnedActor->speedXZ = 2.0f;
+        }
+    }
+}
+
+void func_80B9BD84(ObjTaru* this, PlayState* play) {
+    if (!OBJ_TARU_GET_8000(&this->dyna.actor)) {
+        func_80B9BC64(this, play);
+    } else {
+        func_80B9BCBC(this, play);
+    }
+}
 
-extern ColliderCylinderInit D_80B9C340;
-extern InitChainEntry D_80B9C36C[];
+void ObjTaru_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    ObjTaru* this = THIS;
+    s32 params8000;
 
-extern UNK_TYPE D_06000420;
-extern UNK_TYPE D_06001040;
-extern UNK_TYPE D_06001100;
-extern UNK_TYPE D_06001CB0;
+    DynaPolyActor_Init(&this->dyna, 0);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Taru/func_80B9B6E0.s")
+    if (OBJ_TARU_GET_80(thisx)) {
+        if (Flags_GetSwitch(play, OBJ_TARU_GET_7F(thisx))) {
+            Actor_MarkForDeath(&this->dyna.actor);
+        } else {
+            DynaPolyActor_LoadMesh(play, &this->dyna, &object_taru_Colheader_001CB0);
+        }
+        if (OBJ_TARU_GET_100(&this->dyna.actor)) {
+            this->dyna.actor.scale.x = 0.2f;
+        }
+    } else {
+        params8000 = OBJ_TARU_GET_8000(thisx);
+        DynaPolyActor_LoadMesh(play, &this->dyna, &object_taru_Colheader_000FC8);
+        Collider_InitAndSetCylinder(play, &this->collider, &this->dyna.actor, &sCylinderInit);
+        Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Taru/func_80B9B74C.s")
+        if (params8000 == 0) {
+            if (Item_CanDropBigFairy(play, OBJ_TARU_GET_3F(&this->dyna.actor), OBJ_TARU_GET_7F00(&this->dyna.actor))) {
+                this->unk_1AC = 1;
+                this->dyna.actor.flags |= ACTOR_FLAG_10;
+            }
+        }
+        if ((params8000 != 1) || (!func_80B9B6E0(this, play))) {
+            this->unk_1AD = -1;
+        }
+    }
+    this->actionFunc = func_80B9C07C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Taru/func_80B9B9C8.s")
+void ObjTaru_Destroy(Actor* thisx, PlayState* play) {
+    ObjTaru* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Taru/func_80B9BC64.s")
+    if (!OBJ_TARU_GET_80(thisx)) {
+        Collider_DestroyCylinder(play, &this->collider);
+    }
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Taru/func_80B9BCBC.s")
+s32 func_80B9BF7C(ObjTaru* this) {
+    s32 pad;
+    s32 phi_a3 = 0;
```

Comment:
This looks like a boolean:

```suggestion
    s32 phi_a3 = false;
```